plat/kvm/efi.c

```diff
@@ -76,16 +81,29 @@ static void _uk_efi_crash(void)
 
 #ifdef CONFIG_KVM_BOOT_PROTO_EFI_STUB_DEBUG
 #define UK_EFI_MAX_CRASH_STR_LEN				256
+
+static void uk_efi_printf(const char *str, ...)
+{
+	char fmt_str[UK_EFI_MAX_CRASH_STR_LEN];
+	char str_tmp[UK_EFI_MAX_CRASH_STR_LEN];
+	__s16 str16[UK_EFI_MAX_CRASH_STR_LEN];
+	va_list ap;
+
+	sprintf(fmt_str, "[uk_efi]: %s\r", str);
+
+	va_start(ap, str);
+	vsprintf(str_tmp, fmt_str, ap);
+	va_end(ap);
+
+	ascii_to_utf16(str_tmp, (char *)str16, UK_EFI_MAX_CRASH_STR_LEN - 1);
+	uk_efi_st->con_out->output_string(uk_efi_st->con_out, str16);
+}
+
 /* UEFI for proper \n, we must also use CRLF */
-#define uk_efi_crash(str)						\
-	do {								\
-		__s16 str16[UK_EFI_MAX_CRASH_STR_LEN];			\
-									\
-		ascii_to_utf16("[uk_efi]: "str"\r", (char *)str16,	\
-			       UK_EFI_MAX_CRASH_STR_LEN - 1);		\
-		uk_efi_st->con_out->output_string(uk_efi_st->con_out,	\
-						  str16);		\
-		_uk_efi_crash();					\
+#define uk_efi_crash(str)					\
+	do {							\
+		uk_efi_printf(str);				\
+		_uk_efi_crash();				\
```

Comment:
I know that I initially renamed this with a leading `_`, but now we try to maintain a convention where we do not have such symbols. Let's rename `uk_efi_crash` to `UK_EFI_CRASH` (like `UK_CRASH`) and `_uk_efi_crash` to just `uk_efi_crash` with this opportunity :smile:. Also, shouldn't the signature be with `str, ...` and `uk_efi_printf` have two arguments as well now? :thinking: